vowpalwabbit/slim/include/vw_slim_predict.h

```diff
@@ -13,6 +13,150 @@
 #include "model_parser.h"
 #include "opts.h"
 
+namespace vw_slim { namespace internal {
+  template <typename It1, typename It2>
+  class location_reference;
+
+  template <typename It1, typename It2>
+  class location_value
+  {
+   public:
+    using Val1 = typename std::iterator_traits<It1>::value_type;
+    using Val2 = typename std::iterator_traits<It2>::value_type;
+
+    Val1 _val1;
```

Comment:
Should these be references?

Reply:
They are values used to hold temp copies.

---
In reply to: [310806177](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r310806177) [](ancestors = 310806177)